<launch>
   
    <arg name="use_lidar" default="false"/>

    <!-- Load robot model -->
    <param name="robot_description"
    	   textfile="$(find trikey_description)/urdf/TrikeyBaseAssembly.urdf"/>

    <!-- Load hardware interface -->
    <node name="bumpybot_hardware_interface" pkg="bumpybot_hw_interface" type="bumpybot_hw_main"
          output="screen"/>

    <!-- Load controller config -->
    <rosparam command="load" file="$(find trikey_control)/config/trikey_control.yaml"/>

    <!-- Load controller manager (TODO add joint_state_controller) 
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
          output="screen" ns="/trikey" args="trikey_base_controller"/>
    -->

    <group if="$(arg use_lidar)">
        <!-- Load lidar -->
        <include file="$(find rplidar_ros)/launch/rplidar.launch"/>
    </group>
</launch>
